src/lib.rs

```diff
@@ -716,6 +717,75 @@ impl<K: EnrKey> Enr<K> {
         Ok(())
     }
 
+    /// Removes key/value mappings and adds or overwrites key/value mappings to the ENR record as
+    /// one sequence number update. An `EnrKey` is required to re-sign the record once modified.
+    /// Reverts whole ENR record on error.
+    ///
+    /// Returns the previous values as rlp encoded bytes if they exist for the removed and added/
+    /// overwritten keys.
+    pub fn remove_insert<'a>(
+        &mut self,
+        remove_keys: impl Iterator<Item = impl AsRef<[u8]>>,
+        insert_key_values: impl Iterator<Item = (impl AsRef<[u8]>, &'a [u8])>,
+        enr_key: &K,
+    ) -> Result<(PreviousRlpEncodedValues, PreviousRlpEncodedValues), EnrError> {
+        let enr_backup = self.clone();
+
+        let mut removed = Vec::new();
+        for key in remove_keys {
+            removed.push(self.content.remove(key.as_ref()));
+        }
+
+        // add the new public key
+        let public_key = enr_key.public();
+        self.content.insert(
+            public_key.enr_key(),
+            rlp::encode(&public_key.encode().as_ref()).freeze(),
+        );
+        if self.size() > MAX_ENR_SIZE {
```

Comment:
We could probably remove all these intermediate `self.size()` checks and just keep the one at the end. 

The `size()` function I think has to iterate through each element and RLP encode then check the resulting length, so it might also be better to just check it once at the end and revert if its too large. What do you think?

Reply:
My bad I didn't check what that function computes, it's true it encodes the keys. My idea was to save computation resources but that doesn't really happen then. Thanks for the explanation! 

Why does `self.size()` not take into account the size added by the signature? `self.size()` calls `self.rlp_content()` which unlike the impl of `rap::Encodable` trait doesn't add the signature.
https://github.com/sigp/enr/blob/89cb0c964153e18eb0e296c66b261f83c821aaae/src/lib.rs#L730
https://github.com/sigp/enr/blob/89cb0c964153e18eb0e296c66b261f83c821aaae/src/lib.rs#L840

Reply:
Good point. This sounds like a bug. We should probably adjust the size() function to include the rlp encoded signature. We'd probably have to do the check after we sign the changes, in case in the future some funky signature scheme gives us different sized sigs.